experimental/unitsconversion/src/measureunit.rs

```diff
@@ -0,0 +1,271 @@
+        } else if let Some(part) = part.strip_prefix("yobi") {
+            (80, part)
+        } else {
+            (0, part)
+        }
+    }
+
+    // TODO: consider using a sufficient trie search for finding the unit id.
+    /// Get the unit id.
+    /// NOTE:
+    ///    if the unit id is found, the function will return (unit id, part without the unit id and without `-` at the beginning of the remaining part if it exists).
+    ///    if the unit id is not found, the function will return None.
+    fn get_unit_id<'data>(
+        part: &'data str,
+        trie: &ZeroTrie<ZeroVec<'data, u8>>,
+    ) -> Option<(usize, &'data str)> {
+        // TODO(#4379): this is inefficient way to search for an item in a trie.
+        // we must implement a way to search for a prefix in a trie.
+        let mut result = None;
+        for (index, _) in part.char_indices() {
+            let identifier = &part[..=index];
+            if let Some(value) = trie.get(identifier.as_bytes()) {
+                result = Some((value, &part[identifier.len()..]));
+            }
+        }
+
+        result
+    }
+
+    /// Process a part of an identifier.
+    /// For example, if the whole identifier is: "square-kilometer-per-second",
+    /// this function will be called for "square-kilometer" with sign (1) and "second" with sign (-1).
+    fn analyze_identifier_part(
+        identifier_part: &str,
+        sign: i8,
+        trie: &ZeroTrie<ZeroVec<'_, u8>>,
+    ) -> Option<Vec<MeasureUnitItem>> {
+        let mut identifier = identifier_part;
+        let mut measure_unit_items = Vec::<MeasureUnitItem>::new();
+        while !identifier.is_empty() {
+            let (power, identifier_power) = Self::get_power(identifier);
+            let (si_prefix, identifier_si) = Self::get_si_prefix(identifier_power);
+            let (unit_id, identifier_unit) = Self::get_unit_id(identifier_si, trie)?;
+
+            measure_unit_items.push(MeasureUnitItem {
+                power: power as i8 * sign,
+                si_prefix,
+                unit_id: unit_id as u16,
+            });
+
+            identifier = identifier_unit.get(1..).unwrap_or("");
+        }
+
+        Some(measure_unit_items)
+    }
+
+    // TODO: add test cases for this function.
+    /// Process an identifier.
+    pub fn try_from_identifier<'data>(
+        identifier: &'data str,
+        trie: &ZeroTrie<ZeroVec<'data, u8>>,
+    ) -> Option<Vec<MeasureUnitItem>> {
+        let mut per_split = if identifier.contains("-per-") {
+            identifier.split("-per-")
+        } else {
+            identifier.split("per-")
+        };
+
+        let num_part = per_split.next().unwrap_or("");
+        let den_part = per_split.next().unwrap_or("");
+        if per_split.next().is_some() {
+            return None;
+        }
+
+        let measure_unit_items = [
+            Self::analyze_identifier_part(num_part, 1, trie)?,
+            Self::analyze_identifier_part(den_part, -1, trie)?,
```

Comment:
It's better to allocate a single vector and let `anaylyze_identifier_part` push into it. Currently you're allocating three vectors.